### PR TITLE
fix(Student): Add teammate already in another team error message

### LINKED ISSUE #245: Adding teammate already in a different team should show error message
If a student tries to add a teammate that is already in a different team in the run, there should be an error message but nothing happens and the "Add Teammate" element disappears. This happens if the teammate you are trying to add is a Google user or regular WISE user.

Refer to https://github.com/WISE-Community/WISE-Client/issues/929

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
Check if the user that is being added is already in another workgroup.

## Test
- Test with and use testing instructions in https://github.com/WISE-Community/WISE-Client/pull/1522

Closes #245

### CODE REVIEW

File: src/main/java/org/wise/portal/presentation/web/controllers/student/StudentAPIController.java [hunk 1]
```diff
@@ -407,8 +407,17 @@ HashMap<String, Object> canBeAddedToWorkgroup(Authentication auth,
...
+    Workgroup workgroupOfUserBeingAdded = null;
+    if (workgroupService.isUserInAnyWorkgroupForRun(user, run)) {
+      workgroupOfUserBeingAdded = workgroupService.getWorkgroupListByRunAndUser(run, user).get(0);
+      if (isUserAlreadyInAnotherWorkgroup(workgroup, workgroupOfUserBeingAdded)) {
+        response.put("status", false);
+        return response;
+      }
+    }
+    if (workgroup == null) {
+      workgroup = workgroupOfUserBeingAdded;
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/student/StudentAPIController.java]: The logic in this block is a bit hard to follow. Is there a cleaner way to write this? Would something like this work?
```suggestion
    if (workgroupService.isUserInAnyWorkgroupForRun(user, run)) {
      Workgroup workgroupOfUserBeingAdded = workgroupService.getWorkgroupListByRunAndUser(run, user).get(0);
      if (isUserAlreadyInAnotherWorkgroup(workgroup, workgroupOfUserBeingAdded)) {
        response.put("status", false);
        return response;
      } else {
        workgroup = workgroupOfUserBeingAdded;
      }
    }
```

File: src/main/java/org/wise/portal/presentation/web/controllers/student/StudentAPIController.java [hunk 2]
```diff
@@ -433,4 +442,8 @@ HashMap<String, Object> canBeAddedToWorkgroup(Authentication auth,
     response.put("workgroupMembers", members);
     return response;
   }
+
+  private Boolean isUserAlreadyInAnotherWorkgroup(Workgroup workgroup1, Workgroup workgroup2) {
+    return workgroup1 != null && workgroup2 != null && workgroup1.getId() != workgroup2.getId();
+  }
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/student/StudentAPIController.java]: Should this function return boolean (primitive) since we're only interested in true/false and not the Boolean api? And is there a better name for this function? The function has nothing to do with the user. Something like ```isNotTheSame()```?

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Functionality works, but the new logic in canBeAddedToWorkgroup() is a bit hard to follow. Is there a better way to write it? I added a suggestion inline, but I'm not sure if it's the same logic.

**Comment by geoffreykwan** [comment]
I've cleaned up the code using an exception.

**Review by hirokiterashima — APPROVED** [review]
LGTM. ```canBeAddedToWorkgroup()``` is easier to follow now. 

Let's refactor this class in a future task. The new ```getWorkgroupToAddTo()``` has a cognitive complexity of 10. I'm not sure how, but I think there's got to be a better way to write this- maybe break it down into multiple endpoints?
